Test failures with Wildfly/EAP adapters using TLS

### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

ci

### Describe the bug

When running tests using the Wildfly/EAP adapters with TLS, the tests fail to run.

### Version

latest

### Expected behavior

The tests are executed successfully 

### Actual behavior

```
[ERROR] org.keycloak.testsuite.adapter.example.cors.CorsExampleAdapterTest.angularCorsProductTest  Time elapsed: 5.007 s  <<< ERROR!
java.lang.IllegalStateException: The server should be WildFly (either ManagementProtocol.HTTP(S)_REMOTING was used or the 'creaper.wildfly' system property was set), but client libraries are AS7-only
```

### How to Reproduce?

1. `mvn -f testsuite/integration-arquillian/pom.xml clean install -DskipTests -Pauth-server-quarkus`
2. `mvn -f testsuite/integration-arquillian/pom.xml clean install -DskipTests -Dapp.server.ssl.required=true -Dauth.server.host=auth-server-127-0-0-1.nip.io -Dapp.server.host=127.0.0.1.sslip.io -Papp-server-wildfly`
3. `mvn -f testsuite/integration-arquillian/tests/base/pom.xml clean install -Pauth-server-quarkus -Dapp.server.ssl.required=true -Dauth.server.host=auth-server-127-0-0-1.nip.io -Dapp.server.host=127.0.0.1.sslip.io -Papp-server-wildfly -Dtest=CorsExampleAdapterTest#angularCorsProductTest -Dbrowser=chrome -Dwebdriver.chrome.driver=/usr/bin/chromedriver`

### Anything else?

Wildfly version: 23.0.2.Final

The full stack-trace is the following:

```
[ERROR] org.keycloak.testsuite.adapter.example.cors.CorsExampleAdapterTest.angularCorsProductTest  Time elapsed: 4.146 s  <<< ERROR!
java.lang.IllegalStateException: The server should be WildFly (either ManagementProtocol.HTTP(S)_REMOTING was used or the 'creaper.wildfly' system property was set), but client libraries are AS7-only
	at org.wildfly.extras.creaper.core.online.OnlineOptions.createModelControllerClient(OnlineOptions.java:448)
	at org.wildfly.extras.creaper.core.online.OnlineManagementClientImpl.connect(OnlineManagementClientImpl.java:42)
	at org.wildfly.extras.creaper.core.online.OnlineManagementClientImpl.<init>(OnlineManagementClientImpl.java:38)
	at org.wildfly.extras.creaper.core.online.OnlineOptions$2.create(OnlineOptions.java:534)
	at org.wildfly.extras.creaper.core.ManagementClient.online(ManagementClient.java:34)
	at org.keycloak.testsuite.arquillian.AppServerTestEnricher.getManagementClient(AppServerTestEnricher.java:186)
	at org.keycloak.testsuite.arquillian.AppServerTestEnricher.getManagementClient(AppServerTestEnricher.java:181)
	at org.keycloak.testsuite.arquillian.AppServerTestEnricher.enableHTTPSForAppServer(AppServerTestEnricher.java:280)
	at org.keycloak.testsuite.adapter.AbstractAdapterTest.setUpAppServer(AbstractAdapterTest.java:74)
        ...
```

The problem is that this commit https://github.com/keycloak/keycloak/commit/ef272f7668030b196a2376ec45cc09d9271ff31d removed core version 19 and upgraded to 21 (to create the new saml adapters). But the core version is also used in tests to communicate with the wildfly/eap server and configure things (https for example). When upgrading to 21 there is a method change and resulted in that exception.

I'm going to add a new `tests.wildfly.core.version` var to to be used inside the tests. It should be compatible with the wildfly (or eap) used for tests (wildfly 23.0.2 currently).

As commented to @miquelsi maybe we should stop testing OIDC adapters with wildfly/eap and just test SAML adapters (the only adapters that will remain). That way we can start testing with a newer wildfly version not limited by the OIDC adapters. But for the moment just fixing the tests are they are now.

--- a/pom.xml
+++ b/pom.xml
@@ -48,6 +48,9 @@
 
         <project.build-time>${timestamp}</project.build-time>
 
+        <!-- core version for app-server-wildfly/eap tests, should match wildfly/eap server -->
+        <tests.wildfly.core.version>19.0.1.Final</tests.wildfly.core.version>
+
         <!-- Upstream WildFly Versions -->
         <upstream.wildfly.version>29.0.0.Final</upstream.wildfly.version>
         <upstream.wildfly.build-tools.version>1.2.13.Final</upstream.wildfly.build-tools.version>

--- a/testsuite/integration-arquillian/tests/base/pom.xml
+++ b/testsuite/integration-arquillian/tests/base/pom.xml
@@ -527,6 +527,7 @@
             <properties>
                 <app.server>wildfly</app.server> <!--in case the profile is called directly-->
                 <app.server.skip.unpack>false</app.server.skip.unpack>
+                <wildfly.core.version>${tests.wildfly.core.version}</wildfly.core.version>
             </properties>
             <dependencies>
                 <dependency>
@@ -563,6 +564,7 @@
             <properties>
                 <app.server>eap</app.server> <!--in case the profile is called directly-->
                 <app.server.skip.unpack>false</app.server.skip.unpack>
+                <wildfly.core.version>${tests.wildfly.core.version}</wildfly.core.version>
             </properties>
             <dependencies>
                 <dependency>

